qiskit/visualization/dag_visualization.py

```diff
@@ -26,7 +31,48 @@
 from .exceptions import VisualizationError
 
 
+IMAGE_TYPES = {
+    "canon",
+    "cmap",
+    "cmapx",
+    "cmapx_np",
+    "dia",
+    "dot",
+    "fig",
+    "gd",
+    "gd2",
+    "gif",
+    "hpgl",
+    "imap",
+    "imap_np",
+    "ismap",
+    "jpe",
+    "jpeg",
+    "jpg",
+    "mif",
+    "mp",
+    "pcl",
+    "pdf",
+    "pic",
+    "plain",
+    "plain-ext",
+    "png",
+    "ps",
+    "ps2",
+    "svg",
+    "svgz",
+    "vml",
+    "vmlz",
+    "vrml",
+    "vtx",
+    "wbmp",
+    "xdor",
+    "xlib",
+}
+
+
 @_optionals.HAS_GRAPHVIZ.require_in_call
+@_optionals.HAS_PIL.require_in_call
```

Comment:
This is just inlined from the set in rustworkx. I just copied it across so that we could raise the exception as expected for an invalid image type. You can see the original list here: https://github.com/Qiskit/rustworkx/blob/334d4aeb02ce29d44336d0b3180b5e2addbd7e48/rustworkx/visualization/graphviz.py#L36-L72

Reply:
In that case, whatever we decide about licensing requirements in https://github.com/Qiskit/qiskit/pull/12550#discussion_r1713956498 possibly applies here too.

Reply:
Perhaps, but realistically all I did for this was copy the output of `dot -T?` on my local system several years ago to generate this set. It didn't seem particularly significant to say it's a fork. Arguably the subprocesses and logic pieces are I guess, but I didn't really think it was worth it in this case. It's also kind of moot because I'm the author of all of the code in rustworkx for this.